docs/proposals/2841-flux-hpc/README.md

```diff
@@ -0,0 +1,275 @@
+*   **Limited Scheduling Features:** Kubernetes scheduling does not natively support advanced HPC concepts like custom job queues, graph-based scheduling for complex workflows, or resource reservations that are crucial for managing shared, high-demand computing environments.
+*   **Scheduling Throughput**: Kubernetes is limited by API interactions, and etcd performance. Throughput in standard Kubernetes clusters can range between 10-100 Pods per second, and it can be much higher for HPC workload managers (especially Flux). In Flux, high throughput is enabled via submitting jobs to a hierarchy of Flux instances. We have experiments underway to provide updated throughput numbers for comparison.
+
+By integrating Flux Framework, a next-generation workload manager, we can address these gaps directly. Flux provides a graph-based scheduling model and a robust `zeromq`-based bootstrapping mechanism, offering a superior alternative for demanding distributed jobs.
+
+This KEP proposes a design to integrate Flux into Kubeflow Trainer by extending the `JobSet` backend, providing a seamless user experience for running HPC-style workloads on Kubernetes.
+
+### Goals
+
+1.  **Integrate Flux Framework into Kubeflow Trainer** by creating a new plugin that extends the `JobSet` backend. This plugin will dynamically inject and configure a Flux cluster "on-the-fly" for a given `TrainJob`. While the Flux Operator [MiniCluster](https://github.com/flux-framework/flux-operator) will not be used directly, it's design strategy will be.
+2.  **Provide a Robust, SSH-less MPI Bootstrap Mechanism.** Enable users to run any MPI-enabled application without the overhead of configuring SSH keys or a dedicated MPI user, simply by leveraging Flux.
+3.  **Expose Advanced HPC Scheduling Capabilities.** Lay the groundwork for users to leverage Flux's features, such as fine-grained resource mapping, reservations, hierarchical management, and job queueing, within their Kubeflow training jobs.
+4.  **Introduce an Extensible API Policy.** Define a new `hpcPolicy` that is generic enough to support other HPC workload managers in the future (e.g., Slurm, LSF), with Flux as the initial, reference implementation.
+
+### Non-Goals
+
+1.  **Replace JobSet:** This proposal extends and enhances `JobSet`, not replaces it. `JobSet` remains the core abstraction for managing the group of pods.
+2.  **Support Additional HPC Schedulers Immediately:** The initial implementation will focus exclusively on Flux Framework. Support for other managers can be added later under the same `hpcPolicy` API by respective parties that have interest.
+3.  **Re-implement the MPI Operator:** This proposal provides an alternative to the MPI Operator's launcher/worker model by leveraging Flux's native capabilities, rather than replicating its logic.
+
+## Proposal
```

Comment:
Please can you add a couple of user stories which explain when Flux plugin will be useful, like here: https://github.com/kubeflow/trainer/tree/master/docs/proposals/2437-volcano-scheduler#user-stories